yeat/cli/config.py

```diff
@@ -8,44 +8,46 @@
 # -------------------------------------------------------------------------------------------------
 
 import json
+from pathlib import Path
 from warnings import warn
 
 
-ALGORITHMS = ("spades", "megahit", "unicycler", "flye", "canu")
-KEYS = ["algorithm", "extra_args"]
+KEYS1 = ("samples", "assemblers")
+KEYS2 = ("label", "algorithm", "extra_args", "samples")
+PAIRED = ("spades", "megahit", "unicycler")
+PACBIO = ("canu", "flye")
+ALGORITHMS = PAIRED + PACBIO
 
 
 class AssemblyConfigurationError(ValueError):
     pass
 
 
 class AssemblerConfig:
-    def __init__(self, algorithm, extra_args=""):
-        if algorithm not in ALGORITHMS:
-            raise AssemblyConfigurationError(f"Unsupported assembly algorithm '{algorithm}'")
-        self.algorithm = algorithm
-        self.extra_args = extra_args
+    def __init__(self, data, threads):
+        self.data = data
+        self.validate()
+        self.threads = threads
+        self.create_sample_and_assembler_objects()
+        self.batch()
```

Comment:
The purpose of the `AssemblerConfig` class is to parse the "config" file and create `Sample` and `Assembler` objects based on the config's contents. When the config data is parsed, the data is first validated to ensure that the user entered in all the required entries. If correct, the class creates two list of classes of "samples" and "assemblers". Within each of those objects, further checking is done such as: Does the file exist? Did the user input enough "cores" for a given assembly algorithm? Is the assembly algorithm supported by YEAT?

Once all of the `Sample` and `Assembly` objects are created successfully, the `batch()` is called. The purpose of this function is to create a nested dictionary that are grouped by the algorithm in each `Assembly` object and has its corresponding `Sample` objects associated with them.